### PR TITLE
database_api refactoring and API for blockchain explorer

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**

### CODE REVIEW

File: libraries/plugins/delayed_node/include/scorum/delayed_node/delayed_node_plugin.hpp [hunk 1]
```diff
@@ -55,4 +58,4 @@ class delayed_node_plugin : public scorum::app::plugin
     void sync_with_trusted_node();
 };
 }
-} // scorum::account_history
+} // scorum::blockchain_history
```
Comment (abolinbsu) [inline libraries/plugins/delayed_node/include/scorum/delayed_node/delayed_node_plugin.hpp]: fix namespace

File: tests/chain_tests/plugins/blockinfo_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    operation input_op;
+
+    generate_block();
+
+    {
+        transfer_to_scorumpower_operation op;
+        op.from = alice.name;
+        op.to = bob.name;
+        op.amount = ASSET_SCR(feed_amount / 10);
+        push_operation(op, alice.private_key, false);
```
Comment (alexplevako) [inline tests/chain_tests/plugins/blockinfo_tests.cpp]: flag `false` is not informative, you should create function with informative name.

File: libraries/plugins/blockchain_history/blockchain_history_api.cpp [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+blockchain_history_api::~blockchain_history_api()
+{
+}
+
+void blockchain_history_api::on_api_startup()
+{
+}
+
+std::map<uint32_t, applied_operation>
+blockchain_history_api::get_ops_history(uint32_t from_op, uint32_t limit, const applied_operation_type& opt) const
```
Comment (alexplevako) [inline libraries/plugins/blockchain_history/blockchain_history_api.cpp]: opt is a bad name, please avoid this practice. name should be informative.

File: tests/ucommon/actor.hpp [hunk 4]
```diff
@@ -32,6 +32,11 @@ class Actor
 
     Actor& percent(float prc);
 
+    operator const std::string&() const
```
Comment (alexplevako) [inline tests/ucommon/actor.hpp]: move in cpp

**Pull request merged**